internal/mode/static/nginx/runtime/manager.go

```diff
@@ -17,18 +17,33 @@ import (
 )
 
 const (
-	pidFile            = "/var/run/nginx/nginx.pid"
+	PidFile            = "/var/run/nginx/nginx.pid"
 	pidFileTimeout     = 10000 * time.Millisecond
 	nginxReloadTimeout = 60000 * time.Millisecond
 )
 
 type (
-	readFileFunc  func(string) ([]byte, error)
-	checkFileFunc func(string) (fs.FileInfo, error)
+	ReadFileFunc  func(string) ([]byte, error)
+	CheckFileFunc func(string) (fs.FileInfo, error)
 )
 
 var childProcPathFmt = "/proc/%[1]v/task/%[1]v/children"
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . nginxPlusClient
+
+type nginxPlusClient interface {
+	UpdateHTTPServers(upstream string, servers []ngxclient.UpstreamServer) (added []ngxclient.UpstreamServer, deleted []ngxclient.UpstreamServer, updated []ngxclient.UpstreamServer, err error)
+	GetUpstreams() (*ngxclient.Upstreams, error)
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . processHandler
+
+type processHandler interface {
```

Comment:
gentle ping @sjberman do you want me to add processHandlerImpl ? so I can use that outside, but then 
func(p *ProcessHandlerImpl) FindMainProcess gets also updated, if this is even good approach, the goal is to cover these nil, nil that I used for real implementation

Reply:
Yes, that would be necessary to create an implementation.